File: fibonacci_sum_last_digit.cpp
#include <iostream>
// Find the last digit of a sum of the first n Fibonacci
// numbers. Playing around with small values of n shows that
// the nth sum is equal to F(n+2)-1, and we previously solved how
// to calculate the last digit of F(m) 
int fibonacci_sum(long long n) {
    if (n <= 1)
        return n;
    int tmp_previous;
    int previous = 0;
    int current  = 1;
    
    for (long long i = 0; i < n+1; ++i) {
        tmp_previous = previous;
        previous = current;
        current = (tmp_previous + current)%10;
    }

    return (current-1);
}

int main() {
    long long n = 0;
    std::cin >> n;
    std::cout << fibonacci_sum(n);
}

File: python/fibonacci_sum_last_digit.py
# Uses python3
# Find the last digit of a sum of the first n Fibonacci
# numbers. Playing around with small values of n shows that
# the nth sum is equal to F(n+2)-1, and we previously solved how
# to calculate the last digit of F(m) 

def fibonacci_sum(n):
    if n <= 1:
        return n

    previous = 0
    current  = 1

    for _ in range(n+1):
        previous, current = current, (previous + current)%10

    return (current-1)

n = int(input())
print(fibonacci_sum(n))
